Assets/Scripts/VR/VRControllers/VRControllerManager.cs

```diff
@@ -0,0 +1,304 @@
+            inverseLeftController.controllerTransform.gameObject.SetActive(!value);
+
+            Transform toolsController = GlobalState.Instance.toolsController;
+            Transform paletteController = GlobalState.Instance.paletteController;
+
+
+            // Update controller's displays
+            rightController.controllerDisplay.text = "";
+            inverseRightController.controllerDisplay.text = "";
+            leftController.controllerDisplay.text = "";
+            inverseLeftController.controllerDisplay.text = "";
+
+            // Update tooltips
+            Tooltips.HideAll(VRDevice.PrimaryController);
+            Tooltips.HideAll(VRDevice.SecondaryController);
+            ToolBase tool = ToolsManager.CurrentTool();
+            if (null != tool)
+            {
+                tool.SetTooltips();
+            }
+            GlobalState.Instance.playerController.HandleCommonTooltipsVisibility();
+
+            Transform palette = GlobalState.Instance.paletteController.Find("PaletteHandle");
+            if (value)
+            {
+                SetHolders(leftController, rightController, toolsController, paletteController, palette);
+            }
+            else
+            {
+                SetHolders(inverseRightController, inverseLeftController, toolsController, paletteController, palette);
+            }
+        }
+
+        private void SetHolders(VRController primary, VRController secondary, Transform toolsController, Transform paletteController, Transform palette)
+        {
+            if (null != palette) palette.SetPositionAndRotation(primary.paletteHolder.position, primary.paletteHolder.rotation);
+            toolsController.Find("mouthpieces").SetPositionAndRotation(secondary.mouthpieceHolder.position, secondary.mouthpieceHolder.rotation);
+            toolsController.Find("SelectionHelper").SetPositionAndRotation(secondary.helperHolder.position, secondary.helperHolder.rotation);
+            paletteController.Find("SceneHelper").SetPositionAndRotation(primary.helperHolder.position, primary.helperHolder.rotation);
+        }
+
+        public void SetPaletteHolder(Transform palette)
+        {
+            if (GlobalState.Settings.rightHanded)
+                palette.SetPositionAndRotation(leftController.paletteHolder.position, leftController.paletteHolder.rotation);
+            else
+                palette.SetPositionAndRotation(inverseRightController.paletteHolder.position, inverseRightController.paletteHolder.rotation);
+        }
+
+        private void GetQuestControllersValues()
+        {
+            Transform toolsController = GlobalState.Instance.toolsController;
+            Transform paletteController = GlobalState.Instance.paletteController;
+
+            rightController = new VRController("OculusQuest/right_controller", toolsController);
+            rightController.controllerTransform.gameObject.SetActive(true);
```

Comment:
Maybe toolsController.Find("mouthpieces").position = rightController.mouthpieceHolder.position; is missing here.

And the three functions are very similar but some strings, it would be better to have only one function and to retrieve these strings from a scriptable object for exemple.